### PR TITLE
[KYUUBI #1564] fix Server connectionUrl force to hostname 

### PR DESCRIPTION
**Pull request by JeremyTon**
### _Why are the changes needed?_

Currently, we force it to hostname, this is not right.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala [hunk 1]
```diff
@@ -76,7 +76,11 @@ class KyuubiRestFrontendService(override val serverable: Serverable)
 
   override def connectionUrl: String = {
     checkInitialized()
-    s"${serverAddr.getCanonicalHostName}:$portNum"
+    conf.get(FRONTEND_REST_BIND_HOST) match {
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/server/KyuubiRestFrontendService.scala:79]: shall we also update line 47 in `initialize` ?

**Comment by ulysses-you** [comment]
thank you @JeremyTon for the contribution ! also cc @zwangsheng

**Comment by zwangsheng** [comment]
thanks for the contribution! How about adding a few test cases to this function?

**Comment by yaooqinn** [comment]
https://github.com/apache/incubator-kyuubi/pull/1602 maybe we shall simply follow this PR

**Comment by yaooqinn** [comment]
this is fixed, i will close this

**Pull request closed without merging**